common/v2/features/ReceiveAssets/ReceiveAssets.tsx

```diff
@@ -0,0 +1,255 @@
+import { translateRaw } from 'translations';
+
+import QRCode from './components/QRCode';
+import AccountDropdown from './components/AccountDropdown';
+// Legacy
+import receiveIcon from 'common/assets/images/icn-receive.svg';
+
+const truncate = (children: string) => {
+  return [children.substring(0, 15), '…', children.substring(children.length - 10)].join('');
+};
+
+const isAssetToken = (tokenType: string) => {
+  if (tokenType === 'base') {
+    return false;
+  }
+  return true;
+};
+
+const QRDisplay = styled.div`
+  margin: auto;
+  width: 60%;
+`;
+
+const ReceivePanel = styled(ContentPanel)`
+  width: 100%;
+  min-width: 500px;
+`;
+
+const SLabel = styled.label`
+  margin-bottom: 8px;
+  color: #333333;
+  font-weight: normal;
+`;
+
+const Fieldset = styled.fieldset`
+  margin-bottom: 15px;
+`;
+
+const FieldsetBox = styled.div`
+  padding: 12px 0;
+  background: #f6f8fa;
+  text-align: center;
+`;
+
+const AssetFields = styled.div`
+  margin-bottom: 15px;
+  align-items: center;
+`;
+
+const Divider = styled.div`
+  height: 1px;
+  margin: 30px 0;
+  background: #e3edff;
+`;
+
+const FullWidthInput = styled(Input)`
+  width: 100%;
+`;
+
+const StyledSelect = styled(Select)`
+  width: 100%;
+  border-radius: 0.125em;
+  border: 0.125em solid rgba(63, 63, 68, 0.05);
+  outline: 0 0 0 0.25em rgba(0, 122, 153, 0.65);
+`;
+
+const Amount = styled.div`
+  width: 100%;
+`;
+
+const Asset = styled.div`
+  margin-top: 15px;
+  width: 100%;
+
+  .is-focused {
+    border-color: none;
+    outline: none;
+    box-shadow: 0 0 0 0.25em rgba(0, 122, 153, 0.65);
+  }
+`;
+
+const ErrorMessage = styled.span`
+  color: red;
+`;
+
+export function ReceiveAssets({ history }: RouteComponentProps<{}>) {
+  const { accounts } = useContext(AccountContext);
+  const { assets } = useContext(AssetContext);
+  const network = getNetworkByName(accounts[0].network);
+  const filteredAssets = network
+    ? assets
+        .filter(asset => asset.networkId === network.id)
+        .filter(asset => asset.type === 'base' || asset.type === 'erc20')
+    : [];
+  const assetOptions = filteredAssets.map(asset => ({ label: asset.name, id: asset.uuid }));
+  const [chosenAssetName, setAssetName] = useState(assetOptions[0].label);
+  const selectedAsset = filteredAssets.find(asset => asset.name === chosenAssetName);
+
+  const [requestAddress, setRequestAddress] = useState('');
+
+  const initialValues = {
+    amount: '0',
+    asset: { label: 'Ethereum', id: '7bbf42b1-9275-120b-0d0a-a788abd75ea0' },
+    chainId: network ? network.chainId : 1
+  };
+
+  const isValidAmount = (decimal: number) => (amount: string) =>
+    numberIsNotNegative(+amount) && validDecimal(amount, decimal);
+
+  const validateAmount = (amount: any) => {
+    let error;
+    if (selectedAsset) {
+      const { decimal } = selectedAsset;
+      if (decimal && !isValidAmount(decimal)(amount)) {
+        error = 'Please enter a valid amount';
+      }
+    }
```

Comment:
Prefer form validation with `Yup` https://github.com/jquense/yup

Reply:
Looking at the formik examples, they don't seem to use yup unless performing validation for the entire form. I think it might be better to not use yup if validating only a single field?